src/Client/AzureManaged/DurableTaskSchedulerClientOptions.cs

```diff
@@ -109,6 +114,48 @@ this.Credential is not null
                 metadata.Add("Authorization", $"Bearer {token.Token}");
             });
 
+        grpcConfig.ServiceConfig? serviceConfig = null;
+        if (this.RetryOptions != null)
+        {
+            grpcConfig.RetryPolicy retryPolicy = new grpcConfig.RetryPolicy
+            {
+                MaxAttempts = this.RetryOptions.MaxRetries ?? GrpcRetryPolicyDefaults.DefaultMaxAttempts,
+                InitialBackoff = TimeSpan.FromMilliseconds(this.RetryOptions.InitialBackoffMs ?? GrpcRetryPolicyDefaults.DefaultInitialBackoffMs),
+                MaxBackoff = TimeSpan.FromMilliseconds(this.RetryOptions.MaxBackoffMs ?? GrpcRetryPolicyDefaults.DefaultMaxBackoffMs),
+                BackoffMultiplier = this.RetryOptions.BackoffMultiplier ?? GrpcRetryPolicyDefaults.DefaultBackoffMultiplier,
+                RetryableStatusCodes = { StatusCode.Unavailable }, // Always retry on Unavailable.
+            };
+
+            if (this.RetryOptions.RetryableStatusCodes != null)
+            {
+                foreach (StatusCode statusCode in this.RetryOptions.RetryableStatusCodes)
+                {
+                    // Added by default, don't need to have it added twice.
+                    if (statusCode == StatusCode.Unavailable)
+                    {
+                        continue;
+                    }
+
+                    retryPolicy.RetryableStatusCodes.Add(statusCode);
+                }
+            }
+
+            grpcConfig.MethodConfig methodConfig = new grpcConfig.MethodConfig
+            {
+                Names = { grpcConfig.MethodName.Default },
+                RetryPolicy = retryPolicy,
+            };
+
+            serviceConfig = new grpcConfig.ServiceConfig
+            {
+                MethodConfigs = { methodConfig },
+            };
+        }
+        else
+        {
+            serviceConfig = GrpcRetryPolicyDefaults.DefaultServiceConfig;
```

Comment:
nit: consider doing this assignment on line 117 so that you can drop this `else` condition.

Reply:
Done.